raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1859,15 +1819,15 @@ private void appendBatch(
                 offsetAndEpoch.offset + 1, Integer.MAX_VALUE);
 
             future.whenComplete((commitTimeMs, exception) -> {
-                int numRecords = batch.records.size();
+                int numRecords = batch.records.get().size();
```

Comment:
We should try to avoid blind calls to `get()`. I'm honestly not too sure how this even works for the case of a control batch. Maybe we are just ignoring the error?

Reply:
Good point, I think it may just be ignored somehow. I wrapped this in `if (batch.records.isPresent()) ` for good measure.